### PR TITLE
chore: add clippy::string_slice

### PR DESCRIPTION
**Pull request by joshka**
Fixes: https://github.com/ratatui-org/ratatui/issues/1056

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -96,6 +96,7 @@ rest_pat_in_fully_bound_structs = "warn"
 string_lit_chars_any = "warn"
 string_to_string = "warn"
 use_self = "warn"
+string_slice = "warn"
```
Comment (EdJoPaTo) [inline Cargo.toml:99]: Sort alphabetically

File: src/style/color.rs [hunk 2]
```diff
@@ -321,9 +321,9 @@ impl FromStr for Color {
                             return Err(ParseColorError);
                         }
                         (
-                            u8::from_str_radix(&s[1..3], 16),
-                            u8::from_str_radix(&s[3..5], 16),
-                            u8::from_str_radix(&s[5..7], 16),
+                            u8::from_str_radix(&s.chars().skip(1).take(2).collect::<String>(), 16),
+                            u8::from_str_radix(&s.chars().skip(3).take(2).collect::<String>(), 16),
+                            u8::from_str_radix(&s.chars().skip(5).take(2).collect::<String>(), 16),
```
Comment (EdJoPaTo) [inline src/style/color.rs:326]: This looks like a performance nightmare. It should use the same iterator. And collect String does heap allocations which could be prevented by only using references. 

Maybe something like `s.get(range).ok()?`? When it doesn’t work it’s not a hex Color.

**Review by EdJoPaTo — CHANGES_REQUESTED** [review]

File: src/style/color.rs (hunk 2 above)
Comment (joshka) [reply src/style/color.rs:326]: Makes sense - will fix.

File: src/style/color.rs [hunk 3]
```diff
@@ -316,17 +316,20 @@ impl FromStr for Color {
...
+                            .get(1..3)
+                            .and_then(|v| u8::from_str_radix(v, 16).ok())
+                            .ok_or(ParseColorError)?;
+                        let green = s
+                            .get(3..5)
+                            .and_then(|v| u8::from_str_radix(v, 16).ok())
+                            .ok_or(ParseColorError)?;
+                        let blue = s
+                            .get(5..7)
+                            .and_then(|v| u8::from_str_radix(v, 16).ok())
```
Comment (EdJoPaTo) [inline src/style/color.rs:330]: I would prefer non single letter variable names. Maybe `str`, maybe `blue`?

**Comment by joshka** [comment]
btw @EdJoPaTo I think you were right about that multiple dependencies lint.

```
❯ cargo tree -i windows-targets@0.48.5 --target all
windows-targets v0.48.5
├── parking_lot_core v0.9.9
│   └── parking_lot v0.12.1
│       └── crossterm v0.27.0
│           └── ratatui v0.26.2 (/Users/joshka/local/ratatui)
└── windows-sys v0.48.0
    └── mio v0.8.10
        ├── crossterm v0.27.0 (*)
        └── signal-hook-mio v0.2.3
            └── crossterm v0.27.0 (*)

❯ cargo tree -i windows-targets@0.52.5 --target all
windows-targets v0.52.5
└── windows-sys v0.52.0
    ├── console v0.15.8
    │   └── better-panic v0.3.0
    │       [dev-dependencies]
    │       └── ratatui v0.26.2 (/Users/joshka/local/ratatui)
    ├── is-terminal v0.4.12
    │   └── criterion v0.5.1
    │       [dev-dependencies]
    │       └── ratatui v0.26.2 (/Users/joshka/local/ratatui)
    └── winapi-util v0.1.8
        ├── same-file v1.0.6
        │   └── walkdir v2.5.0
        │       └── criterion v0.5.1 (*)
        └── walkdir v2.5.0 (*)
```

**Comment by joshka** [comment]
I think this was probably done well enough elsewhen.

**Pull request closed without merging**